feat: ridge and pca

robincourant opened the pull request.

yakoubix committed:
pca reduction commit

yakoubix committed:
new merge of pca-reduction and data-prep

antoinelerosey committed:
adding the feature selection function using ridge regression method

yakoubix committed:
ReadMe modified with documentation on PCR

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+import pandas as pd
+import numpy as np
+#import sys
```
Remove these 2 lines.

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+import pandas as pd
```
- First: native python modules (ie: all modules [here](https://docs.python.org/3/py-modindex.html)) then, newline, then, installed modules as `pandas` etc, then, newline and import custom module as `data_preparation`
- Sort subgroups in alphabetical order modules

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+from sklearn.linear_model import Ridge
+from sklearn.feature_selection import SelectFromModel
+
+def feature_selection_ridge(dataset:pd.DataFrame):
```
Your method should begin with a verb ex: `select_features_ridge`.

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+
+def feature_selection_ridge(dataset:pd.DataFrame):
+    """
+    Function to select the relevant features to build a model
```
- Documenting functions is done this way : https://stackoverflow.com/questions/3898572/what-is-the-standard-python-docstring-format, using the reST format
- In a docstring, a sentence begins with an upper case and ends with a point (a point is missing at the end)

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+    Function to select the relevant features to build a model
+    """
+    data_work = load_dataset(dataset)
+    prepare_data(data_work) # Perform data preparation before coming up with PCA
```
Comments should be inserted in the line above.

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+    prepare_data(data_work) # Perform data preparation before coming up with PCA
+    nb_columns = len(data_work.columns)
+    
+    #splitting label column from input column
```
Add an upper case and a whitespace after `#`.

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+    label_column=data_work.iloc[:,-1] 
+    data_work=data_work.iloc[:,:-1]
+
+    #Using Ridge regression to select the relevant
```
Add a whitespace.

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,25 @@
+    data_work=data_work.iloc[:,:-1]
+
+    #Using Ridge regression to select the relevant
+    sel_=SelectFromModel(Ridge())
```
- Add whitespaces between operators (here `sel_ = SelectFromModel(Ridge())`
- Give an explicit name to the variable `sel_`

robincourant commented on pca_reduction.py:
```diff
@@ -0,0 +1,23 @@
+import pandas as pd
```
This file should be merged with `feature_engineering`.

robincourant commented on pca_reduction.py:
```diff
@@ -0,0 +1,23 @@
+from sklearn.decomposition import PCA
+
+
+def reduce_dimensionality(dataset: pd.DataFrame):
```
Change this method name according to your method (ex: select_feature_pca`)

robincourant replied:
Add a return hint (ie: what type of object your method returns). Here you should add: `-> pd.DataFrame`.

robincourant reviewed (commented):
- Remove the two `.pyc` files (check `.gitignore` file)
- Resolve threads pointed out in the comments

robincourant commented on pca_reduction.py:
```diff
@@ -0,0 +1,23 @@
+    data_transformed = pca.fit_transform(data_work)
+    data_transformed=pd.DataFrame(data=data_transformed,columns=[f'principal components {i}' for i in range (np.shape(data_transformed)[1])]) # Concatenating transformed columns into one dataframe.
+    data_transformed['label'] = label_column # appending column label to the transformed dataframe at the last position.
+    return data_transformed
```
Add a newline at the end of the file.

yakoubix committed:
Modified code with threads resolved

antoinelerosey committed:
Adding the features selection using the correlation method

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,55 @@
+import numpy as np
```
In your case you should have:
```
import numpy as np
import pandas as pd
from sklearn.feature_selection import SelectFromModel, SelectKBest,f_regression
from sklearn.linear_model import Ridge

from data_preparation import load_data, prepare_data
```

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,55 @@
+from data_preparation import *
+
+def select_features_ridge(filename:str):
+    """Select the relevant features using Ridge Regression method.
```
You should have: 

```
def select_features_ridge(filename:str) -> pd.Index:
"""Select the relevant features using Ridge Regression method.

:param filename: path to the dataset's '.csv' file.
:return: the selected features.
"""
```

Correct the others.

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,55 @@
+    nb_columns = len(data_work.columns)
+    
+    # Splitting label column from input column
+    label_column=data_work.iloc[:,-1] 
```
Don't forget whitespaces around operators.

robincourant commented on feature_engineering.py:
```diff
@@ -0,0 +1,55 @@
+    return selected_features
+
+def select_features_correlation(filename:str,number_of_feature_to_keep:int):
+    """Select the relevant features by looking at correlation
```
Newline and point are missing.

iqdecay reviewed (changes requested):
Git commits should be made according to this : https://hackernoon.com/git-it-together-some-tips-on-commit-etiquette-and-best-practices-for-junior-developers-1f147b8dfd56

antoinelerosey committed:
review modification

iqdecay committed:
Merge 'pca_reduction' into 'feature_selection'

- This merge will help prevent duplicate code

robincourant committed:
Refactor 'feature-engineering'.

- Merge 'pca-reduction' and 'feature-engineering' in 
'feature-engineering'
- Delete 'pca-reduction'
- Change the input of feature selection methods: now it's loaded 
datasets (remove the data prep inside methods)
